main/guides/integration/name-services.md

```diff
@@ -0,0 +1,106 @@
+
+## agoricNames / agoricNamesAdmin - well known names
+
+To look up well-known objects, such as the BLD brand:
+
+```js
+const brandHub = E(agoricNames).lookup('brand');
+const bldBrand = await E(brandHub).lookup('BLD');
+```
+
+or, equivalently:
+
+```js
+const bldBrand = await E(agoricNames).lookup('brand', 'BLD');
+```
+
+`agoricNames` is a _NameHub_. In general, `E(hub).lookup(key1, key2, ...rest)` is equivalent
+to `E(E(hub).lookup(key1)).lookup(key2, ...rest)`. The NameHub interface also includes `Map`-like
+methods `has`, `entries`, `keys`, and `values`.
+
+The keys of `agoricNames` include:
+ - `brand` - IST, BLD, Invitation (Zoe invitation brand), timer (a TimerBrand, not an ERTP brand)
+ - `issuer` - IST, BLD, Invitation, ...
+ - `vbankAsset` - IST, BLD, USDC, ATOM, ... - including info about corresponding cosmos denom
+ - `oracleBrand` - USD, ATOM
+ - `installation` - psm, reserve, priceAuthority, ... (see [deployed contracts](../zoe/actual-contracts/))
+ - `instance` - reserve, one for each psm, one for each priceAuthority, ... (see [deployed contracts](../zoe/actual-contracts/))
+
+Adding or updating an entry requires access to the corresponding _NameAdmin_ facet: `agoricNamesAdmin` or
+one of its descendants - most likely as part of [permissioned contract deployment](../coreeval/). For example:
+
+```js
+const { instance } = await E(zoe).startInstance(myNewContract);
+const { brands: { Thing: thingBrand } } = await E(zoe).getTerms(instance);
+const brandAdmin = E(agoricNamesAdmin).lookupAdmin('brand');
+await E(brandAdmin).update('Thing', thingBrand);
+```
+
+### agoricNames in vstorage
+
+The `NameAdmin` interface has an `onUpdate` method for registering a callback.
+This is used to reflec the brand entries, that is: `E(E(agoricNames).lookup('brand')).entries()`
+into vstorage at `published.agoricNames.brand` each time they change.
+The entries for `instance` and the other keys of `agoricNames` are likewise reflected under `published.agoricNames`.
+See [marshalling amounts and instance](../getting-started/contract-rpc.md#marshalling-amounts-and-instances)
+for details on the format of the data stored in vstorage.
+
+_TODO: IOU a screenshot of the vstorage viewer_
+
+## namesByAddress / namesByAddressAdmin, and depositFacet - per account namespace
+
+Each time a smart wallet is provisioned for address `a`, a NameHub / NameAdmin pair is
+installed at `E(namesByAddress).lookup(a)` and `E(namesByAddressAdmin).lookupAdmin(a)`.
+Also, a _deposit facet_ is installed at `E(namesByAddress).lookup(a, 'depositFacet')` so
+that you can send a payment, such as an invitation, to such a smart wallet using
+the `receive` method of the deposit facet:
+
+```js
+const bobAddress = 'agoric1...';
+const anInvitationPayment = await zcf.makeInvitation(...);
+const bobDepositFacet = E(namesByAddress).lookup(bobAddress, 'depositFacet');
+const amtReceived = await E(bobDepositFacet).receive(anInvitationPayment);
+```
```

Comment:
ideally, these would be tested snippets of an example dapp

But dapp-offer-up doesn't use namesByAddress.